docs/gitbook/chain-management/cascading-deployment.md

```diff
@@ -0,0 +1,85 @@
+# Cascading Deployment
+
+
+<div class="tips info">
+  <p><span></span>Coming soon...</p>
+  <p>The cascading release managed by a root node with a set of adming keys is not available yet.</p>
+</div>
+
+Managing an entire blockchain system that exists on many nodes with different key pairs with chains that join in myriad ways can be a headache to manage. Cascading deployment offers a way to maintain a blockchain network from a single point of change.
+
+There are several important components used to implement cascading deployment in an Interbit blockchain network.
+
+## The Interbit Configuration File
+
+The configuration file is a js file that contains intstructions on generating a collection of build artifacts.
+Contains:
+- expressions for resolving variables for:
+  - pubkeys /  chainIds (basically permission resolvers)
+  - chain Ids
+  - covenant hashes
+  - peer lists
+  - folder containing index.html and other static assets to be hosted.  index.html will get altered by the build step to include chainIds
+  - array of supporting static files that should be hosted statically
+
+- composition instructions for
+  - joins
+  - mapping of chains to covenants
+  - tree of parent child relationships
+
+Config files are javascript so that you can use env files and other javascript functions such as importing other config files and composing many.  Consider the case of platform development - a top level config file could run all the covenants across the whole platform as a single interbit hypervisor in dev mode.
+
+Running `interbit-cli build` for a configuration file will generate an Interbit manifest file and a set of corresponding artifacts.
+
+## The Interbit Manifest File
+
+The Manifest file contains statements about how to deploy the build artifacts to a specific environment.  These files can be used to upgrade an existing environment.
+
+The files contain no unresolved aliases or inconsistencies between aliases.  Basically, there are no variables in a manifest - everything is resolved to either a chainId, or a covenantHash, or a public key.
+
+Manifest files are always named after the input config file.
+
+Manifest files are json because they contain no computation, and only act as resolution for variables.  They are statements.
+
+In addition to resolving instructions from a configuration file about your configured chains, the manifest file adds deployment instructions for a root node that manages upgrading your blockchain network should you deploy an upgraded manifest in the future.
+
+## The Root Node
+
+The root node contains a copy of the deployed manifest file and a write join to each of your high level chains. The high level chains in turn contain a write join to their children etc. This write join allows your root node to forward updates to the manifest to your chains should it receive an update. These updates will be forwarded by interbit down the line to child chains until all network updates have been applied.
+
```

Comment:
'to your chains when the manifest is updated'
'update will be cascaded to child chains'